include/fsm/fill_spill_merge.hpp

```diff
@@ -560,8 +568,9 @@ static dh_label_t OverflowInto(
   //If we've got here we have a neighbour, but we couldn't stash water in the
   //neighbour because it was full. So we need to see if our parent knows about
   //us.
-  if(this_dep.parent!=OCEAN && pdep.water_vol==0)
-    pdep.water_vol += this_dep.water_vol + odep.water_vol;
+  if(this_dep.parent!=OCEAN && pdep.water_vol==0 && (pdep.lchild == this_dep.dep_label || pdep.rchild == this_dep.dep_label) && (pdep.lchild == odep.dep_label || pdep.rchild == odep.dep_label))
```

Comment:
Why the conditional?

Reply:
Again, I'm concerned about ocean-linked depressions.